### PR TITLE
document the state backend options for deploying K8s with Operator

### LINKED ISSUE #799: Document k8s + Azure blog deployment
Currently, we already support using Azure blob storage as the object storage type in [operator](https://github.com/risingwavelabs/risingwave-operator/pull/404). We should update the document.

### LINKED ISSUE #1057: Document configuring the Kubernetes cluster to use GCS
https://www.risingwave.dev/docs/current/risingwave-kubernetes/#deploy-a-risingwave-instance

Parallel to s3, minio and hdfs.

### PR DESCRIPTION
**Pull request by hengm3467**
## Info

- **Description**

  - [ What's changed? Which parts of the docs are affected? ]

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  - [ Provide a link to the relevant code PR here, if applicable. ]

- **Related doc issue**
  
  Resolves #799 and #1057

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: docs/deploy/risingwave-kubernetes.md [hunk 1]
```diff
@@ -114,74 +114,202 @@ RisingWave uses etcd for persisting data for meta nodes. It's important to note
...
+curl https://raw.githubusercontent.com/risingwavelabs/risingwave-operator/main/docs/manifests/<sub-directory> -o risingwave.yaml
+```
 
-    1. Replace the URL in the command with the local file path of the modified manifest file and then run the command. For example:
+And then, apply the manifest file by using the following command:
 
-        ```shell
+ ```shell
         kubectl apply -f a.yaml      # relative path
         kubectl apply -f /tmp/a.yaml # absolute path
```
Comment (arkbriar) [inline docs/deploy/risingwave-kubernetes.md]: nit: unnecessary tabs

File: docs/deploy/risingwave-kubernetes.md [hunk 2]
```diff
@@ -114,74 +114,202 @@ RisingWave uses etcd for persisting data for meta nodes. It's important to note
...
+```
 
-    </details>
+To customize the state backend of your RisingWave cluster, edit the `spec:stateStore` section under the RisingWave resource (`kind: RisingWave`).
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs groupId="storage_selection">
+<TabItem value="s3" label="etcd+S3">
```
Comment (arkbriar) [inline docs/deploy/risingwave-kubernetes.md]: nit: better align the names, e.g., `AWS S3`

File: docs/deploy/risingwave-kubernetes.md [hunk 3]
```diff
@@ -114,74 +114,202 @@ RisingWave uses etcd for persisting data for meta nodes. It's important to note
...
 
 </TabItem>
 <TabItem value="minio" label="etcd+MinIO">
```
Comment (arkbriar) [inline docs/deploy/risingwave-kubernetes.md]: nit: ditto. I'd prefer `MinIO`

File: docs/deploy/risingwave-kubernetes.md [hunk 4]
```diff
@@ -114,74 +114,202 @@ RisingWave uses etcd for persisting data for meta nodes. It's important to note
...
+        secretName: gcs-credentials
+        
+        # Key of the service account credentials in the secret.
+        serviceAccountCredentialsKeyRef: ServiceAccountCredentials
+        
+        # Optional, set it to true when the credentials can be retrieved.
+        # useWorkloadIdentity: true
 ```
 
 </TabItem>
```
Comment (arkbriar) [inline docs/deploy/risingwave-kubernetes.md:371]: The output part isn't refactored.

image

**Review by arkbriar — COMMENTED** [review]
BTW, shall we have a paragraph above the state store settings to teach users how to configure the meta store? It should be quite straight forward. There's only one choice `etcd` for now.

**Comment by hengm3467** [comment]
@arkbriar
But I'm not sure if other options have been tested. If so, then adding one section would definitely help. Could you please help summarize that? Thanks!

**Comment by arkbriar** [comment]
> @arkbriar But I'm not sure if other options have been tested. If so, then adding one section would definitely help. Could you please help summarize that? Thanks!

MinIO/S3/S3 compatible/GCS/OSS should have been tested. IMO, we just need to replace the tabs of the outputs into an example of the output. There isn't much difference in the outputs.

File: docs/deploy/risingwave-kubernetes.md (hunk 4 above)
Comment (hengm3467) [reply docs/deploy/risingwave-kubernetes.md:371]: Right. But maybe we don't need to have separate tabs for all configurations. We can simplify the presentation by just having ONE sample output.

**Pull request merged**